extensions/desktop-common/desktop-utils.sh

```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
+
+function prepend_dir() {
+  local var="$1"
+  local dir="$2"
+  if [ -d "$dir" ]; then
+    eval "export $var=\"\$dir\${$var:+:\$$var}\""
+  fi
+}
+
+function append_dir() {
+  local var="$1"
+  local dir="$2"
+  if [ -d "$dir" ]; then
+    eval "export $var=\"\${$var:+\$$var:}\$dir\""
+  fi
+}
+
+function can_open_file() {
+  head -c0 "$1" &> /dev/null
+}
+
+function is_subpath() {
+  dir="$(realpath "$1")"
+  parent="$(realpath "$2")"
+  [ "${dir##$parent/}" != "$dir" ] && return 0 || return 1
+}
+
+# TMPDIR is declared by the snap runtime
```

Comment:
If this gets reworked, consider separating the code from the functions and moving all functions to the top.

Reply:
this is now gone